### PR TITLE
[docs] Revamp docs on observability for ray cluster apps

### PR DESCRIPTION
**Pull request by stephanie-wang**
Signed-off-by: Stephanie Wang <swang@cs.berkeley.edu>

## Why are these changes needed?

Various cleanups around docs on Ray cluster "Monitoring and observability". After #27723, we will move these to a common page outside of VMs/k8s subsections:
1. Add links to the more comprehensive observability section.
2. Move and clean up cluster-specific content from Prometheus metrics to the new Ray Cluster page. I also modified a bunch of text here because previously we were not very clear about what the recommended approach was.
3. Include more specific instructions about setting up observability tools for VMs vs k8s.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by scottsun94** [comment]
What's the relationship between this one and the standalone Observability section?

Screen Shot 2022-08-09 at 7 12 52 PM

**Comment by stephanie-wang** [comment]
> What's the relationship between this one and the standalone Observability section?
> 
> Screen Shot 2022-08-09 at 7 12 52 PM

It's meant to supplement that page by explaining what should be done on a Ray Cluster to set up observability tools. See this line of text:
```
The rest of this page will focus on how to access these services when running a Ray Cluster.
```

**Review by richardliaw — APPROVED** [review]
did a readthrough and it seems fine!

**Comment by richardliaw** [comment]
merge in master to fix linkcheck?

File: doc/source/cluster/cluster_under_construction/running-applications-on-ray-clusters/monitoring-and-observability.rst [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    Execute a command on the cluster using ``ray exec``:
+
+    .. code-block:: shell
+
+        $ ray exec <cluster config file> "ray status"
+
+.. tabbed:: If using Kubernetes
+
+    Execute a command on the cluster using ``kubectl`` and the configured
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/running-applications-on-ray-clusters/monitoring-and-observability.rst:76]: nit
``kubectl`` -> ``kubectl exec``

File: doc/source/cluster/cluster_under_construction/running-applications-on-ray-clusters/monitoring-and-observability.rst [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    RayCluster name. We will use the Service targeting the Ray head pod to
+    execute a CLI command on the cluster.
+
+    .. code-block:: shell
+
+        $ kubectl exec <RayCluster name>-head-svc "ray status"
```
Comment (DmitriGekhtman) [inline doc/source/cluster/cluster_under_construction/running-applications-on-ray-clusters/monitoring-and-observability.rst:82]: To exec commands, you need supply the name of a pod. Instructions for this bit go
 
First identify the name of the head pod  
```kubectl get pod | grep <RayCluster name>-head```

Then run

```
kubectl exec <head pod> -- ray status
```

**Review by cadedaniel — APPROVED** [review]
LGTM from a structural perspective!

**Comment by stephanie-wang** [comment]
Docs-only changes, not waiting for windows tests.

**Pull request merged**